pkg/retrieval/retrieval.go

```diff
@@ -112,134 +112,88 @@ func (s *Service) RetrieveChunk(ctx context.Context, addr swarm.Address, origin
 		flightRoute = addr.String() + originSuffix
 	}
 
-	// topCtx is passing the tracing span to the first singleflight call
-	topCtx := ctx
```

Comment:
the context handling here is very delicate and fragile. why is the change needed?